MailSenderTask2/src/main/java/com/demo/SmtpMailSender.java

```diff
@@ -0,0 +1,37 @@
+package com.demo;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMessageHelper;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SmtpMailSender {
+	
+	private static final String CRON = "*/10 * * * * *";
+	@Autowired
+	private JavaMailSender javaMailSender;	
+	
+	@Scheduled(cron = CRON)
```

Comment:
When you use `@Scheduled` you're telling Spring to call this method automatically. Then where it's going to pickup arguments for execution? Besides that, you still using it directly in command line runner. It means this component will be executed automatically and manually by you. It can produce unexpected results